mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -378,10 +379,18 @@ pub async fn create_wallet_db(
                 .map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
             init_wallet_db(&db).map_to_mm(|err| ZcoinClientInitError::ZcashDBError(err.to_string()))?;
 
+            // check if no sync_params was provided and continue syncing from last height in db if it's > 0.
+            let active_sync_block = min_sync_height.unwrap_or(0) > 0 && sync_params.is_none();
             // Check if the initial block height is less than the previous synchronization height and
             // Rewind walletdb to the minimum possible height.
-            if db.get_extended_full_viewing_keys()?.is_empty() || init_block_height != min_sync_height {
-                info!("Older/Newer sync height detected!, rewinding walletdb to new height: {init_block_height:?}");
+            if db.get_extended_full_viewing_keys()?.is_empty()
+                || (!active_sync_block && init_block_height != min_sync_height)
+            {
+                // let user know we're clearing cache and resyncing from new provided height.
+                if min_sync_height.unwrap_or(0) > 0 && sync_params.is_some() {
```

Comment:
do we need the additional check that `sync_params.is_some()` here?

Reply:
removed